### PR TITLE
Added scintillator information to RecHitTools

### PR DESCRIPTION
**Pull request by jkiesele**
#### PR description:

This PR implements a new feature for the RecHitTools to conveniently access the size of individual HGCAL scintillator cells through their detID. While in the current geometry the size is constant for all cells, this implementation avoids hard-coding the size and is transparent w.r.t. geometry changes.
This feature is useful to study shower densities on truth level, and possibly for studying clustering algorithms.

The addition was discussed in the HGCAL DPG meeting: https://indico.cern.ch/event/1055901/ 

#### PR validation:

The code has been validated to produce the right sizes in eta and phi. It is also stable against wrong detID inputs.
Since it implements a new feature, there are no downstream dependencies.

### CODE REVIEW

**Comment by jkiesele** [comment]
@rovere , @pfs , @cseez

**Comment by perrotta** [comment]
please test

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 1]
```diff
@@ -219,6 +219,16 @@ int RecHitTools::getSiThickIndex(const DetId& id) const {
...
 }
 
+std::pair<float, float> RecHitTools::getScintDEtaDPhi(const DetId& id) const {
+  if (!isScintillator(id)) {
+    LogDebug("getScintDEtaDPhi::InvalidSintDetid")
+        << "det id: " << std::hex << id.rawId() << std::dec << ":" << id.det() << " is not HGCal scintillator!";
+    return std::pair<float, float>(0., 0.);
+  }
+  auto geom = static_cast<const HGCalGeometry*>(getSubdetectorGeometry(id));
+  return std::pair<float, float>(geom->getGeometry(id)->etaSpan(), geom->getGeometry(id)->phiSpan());
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc:229]: why is the cast needed here?
`getGeometry` together with`etaSpan` and `phiSpan`  are all virtual.

```suggestion
  auto cellGeom = getSubdetectorGeometry(id)->getGeometry(id);
  return {cellGeom->etaSpan(), cellGeom->phiSpan()};
```

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 2]
```diff
@@ -219,6 +219,16 @@ int RecHitTools::getSiThickIndex(const DetId& id) const {
   return thickIndex;
 }
 
+std::pair<float, float> RecHitTools::getScintDEtaDPhi(const DetId& id) const {
+  if (!isScintillator(id)) {
+    LogDebug("getScintDEtaDPhi::InvalidSintDetid")
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc:224]:
```suggestion
    LogDebug("getScintDEtaDPhi::InvalidScintDetid")
```

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 3]
```diff
@@ -219,6 +219,16 @@ int RecHitTools::getSiThickIndex(const DetId& id) const {
...
+        << "det id: " << std::hex << id.rawId() << std::dec << ":" << id.det() << " is not HGCal scintillator!";
+    return std::pair<float, float>(0., 0.);
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc:226]:
```suggestion
    return {0.f, 0.f};
```

**Comment by slava77** [comment]
> Reco comparison results: 5155 differences found in the comparisons

this seems related to the discussion in https://github.com/cms-sw/cmsdist/pull/7112#issuecomment-876462874

@makortel @davidlange6 
does it confirm now that the python list order is no longer repeatable ?

**Comment by makortel** [comment]
> does it confirm now that the python list order is no longer repeatable ?

It certainly confirms that something unexpected is going on. I'll try to investigate as well.

**Comment by felicepantaleo** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34391	3046eb5 
- code changes are in line with the PR description and the follow up review
    - the code changes look generally OK (without an explicit use case)
- jenkins tests pass and comparisons with the baseline show no (relevant) differences, since the new code is not used yet
    - differences in reco comparisons are related to the status of the PAT/miniAOD jet tagging code in the IB and will be fixed after #34409 is merged in a reference IB

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**